homeassistant/components/withings/__init__.py

```diff
@@ -103,33 +104,28 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Withings from a config entry."""
-    config_updates: dict[str, Any] = {}
-
-    # Add a unique id if it's an older config entry.
-    if entry.unique_id != entry.data["token"]["userid"] or not isinstance(
-        entry.unique_id, str
-    ):
-        config_updates["unique_id"] = str(entry.data["token"]["userid"])
-
-    # Add the webhook configuration.
-    if CONF_WEBHOOK_ID not in entry.data:
-        webhook_id = webhook.async_generate_id()
-        config_updates["data"] = {
-            **entry.data,
-            **{
-                const.CONF_USE_WEBHOOK: hass.data[DOMAIN][const.CONFIG][
-                    const.CONF_USE_WEBHOOK
-                ],
-                CONF_WEBHOOK_ID: webhook_id,
-            },
+    if CONF_USE_WEBHOOK not in entry.options:
+        new_data = entry.data.copy()
+        new_options = {
+            CONF_USE_WEBHOOK: new_data.get(CONF_USE_WEBHOOK, False),
         }
+        title = new_data.pop("profile")
+        unique_id = str(entry.unique_id)
+        if CONF_WEBHOOK_ID not in new_data:
+            new_data[CONF_WEBHOOK_ID] = async_generate_id()
 
-    if config_updates:
-        hass.config_entries.async_update_entry(entry, **config_updates)
+        hass.config_entries.async_update_entry(
+            entry, title=title, data=new_data, options=new_options, unique_id=unique_id
+        )
+    use_webhook = hass.data[DOMAIN][CONFIG][CONF_USE_WEBHOOK]
+    if use_webhook is not None and use_webhook != entry.options[CONF_USE_WEBHOOK]:
+        new_options = entry.options.copy()
+        new_options |= {CONF_USE_WEBHOOK: use_webhook}
+        hass.config_entries.async_update_entry(entry, options=new_options)
```

Comment:
Can we combine all changes into one update of the config entry?

Reply:
The second one runs everytime. In the coming 2 PRs I am planning to remove this again as this will be set up via the options flow instead of yaml

Reply:
Ok.